mediapipe/tasks/ios/components/containers/sources/MPPLandmark.m

```diff
@@ -0,0 +1,105 @@
+// Copyright 2023 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "mediapipe/tasks/ios/components/containers/sources/MPPLandmark.h"
+
+static const float kFloatDifferenceTolerance = 1e-6f;
+
+@implementation MPPLandmark
+
+- (instancetype)initWithX:(float)x
+                        y:(float)y
+                        z:(float)z
+               visibility:(NSNumber *)visibility
+                 presence:(NSNumber *)presence {
+  self = [super init];
+  if (self) {
+    _x = x;
+    _y = y;
+    _z = z;
+    _visibility = visibility;
+    _presence = presence;
+  }
+  return self;
+}
+
+- (NSUInteger)hash {
```

Comment:
@schmidt-sebastian I have implemented hash similar to the Java implementation. Please note that we are converting the float into an `NSNumber` during computation of hash as apple does not provide a hash function for basic types like `float`.  If we don't perform this conversion, we will have to perform the underlying hashing logic of `NSNumber` which would be [something like this](https://stackoverflow.com/a/26433160).
since `isEqual` compares with a tolerance, we can't guarantee that for equal objects hash will be the same. Docs mention that [hash values of two equal objects should be same if it is intended to be used with hash tables](https://developer.apple.com/documentation/objectivec/1418956-nsobject/1418859-hash). 
But I guess even Java, Python etc., have the same requirement. So I am assuming this is okay.